doc/sphinx/docs/python/tutorials/using_algorithm.rst

```diff
@@ -0,0 +1,62 @@
+.. py_tutorial_using_algorithm
+
+Use of the class :class:`~pygmo.core.algorithm`
+===============================================
+
+The :class:`~pygmo.core.algorithm` class represents a generic optimization
+algorithm. The user codes the details of such an algorithm in a separate class (the
+user-defined algorithm, or UDA) which is then passed to :class:`~pygmo.core.algorithm`
+that provides a single unified interface.
+
+.. note::  The User Defined Algorithms (UDAs) are optimization algorithms (coded by the user) that can
+           be used to build a pygmo object of type :class:`~pygmo.core.algorithm`
+
+Some UDAs (optimization algorithms) are already provided with pygmo and we refer to them as pygmo UDAs.
+
+For the purpose of this tutorial we are going to use a pygmo UDA called :class:`~pygmo.core.cmaes`
+to show the basic construction of a :class:`~pygmo.core.algorithm`, but the same logic would also
+apply to a custom UDAs, that is a UDA that is actually coded by the user.
+
+Let us start:
+
+.. doctest::
+
+    >>> import pygmo as pg
+    >>> algo = pg.algorithm(pg.cmaes(gen = 100, sigma0=0.3))
+    >>> print(algo) #doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+    Algorithm name: CMA-ES: Covariance Matrix Adaptation Evolutionary Strategy [stochastic]
+    Extra info:
+    	Generations: 100
+    	cc: auto
+    	cs: auto
+    	c1: auto
+    	cmu: auto
+    	sigma0: 0.300000
+    	Stopping xtol: 0.000001
+    	Stopping ftol: 0.000001
+    	Memory: 0
+    	Verbosity: 0
+    	Seed: ...
+
+In the code above, after the trivial import of the pygmo package, we define a variable algo
+by constructing a :class:`~pygmo.core.algorithm` from :class:`~pygmo.core.cmaes`, our implementation
+of the Covariance Matrix Adaptation Evolutionary Strategy. To construct the pygmo UDA we also pass
+some parameters (gen and sigma0) whose meaning is documented in :class:`~pygmo.core.cmaes`.
+In the following line we inspect the :class:`~pygmo.core.algorithm`. We can see, at a glance, the
+name of the :class:`~pygmo.core.algorithm` and some extra info that indicate the user (in this case us),
+has implemented, in the UDA (in this case :class:`~pygmo.core.cmaes`), the optional method
+get_extra_info that prints to screen some fundamental parameters defining the UDA.
+
+We may also get back a copy of the UDA, and thus access also its methods not exposed in the
```

Comment:
Right now ``extract()`` gets a copy, but the intention was to change it to return a reference, so we can call mutable methods on it as well (in the same vein we want to change population to return a ref to the internal problem with the getter).